src/drivers/intel/dw-dma.c

```diff
@@ -320,16 +320,9 @@ static int dw_dma_channel_get(struct dma *dma, int req_chan)
 	uint32_t flags;
 	int i;
 
-	if (req_chan >= dma->plat_data.channels) {
```

Comment:
dont remove this check. channels are stored as an array in the DM data so removing this check could meant an array bounds error.

Reply:
> dont remove this check. channels are stored as an array in the DM data so removing this check could meant an array bounds error.

For dw-dma, req_chan is useless. We don't use it for anything. We use DW_MAX_CHAN (which is 8) to make sure we will not exceed the array size.